index.js

```diff
@@ -660,8 +653,11 @@ class IHydraExpress extends HydraExpress {
   */
   init(config, version, registerRoutesCallback) {
     let inner = {};
-    if (version) {
-      inner.version = version;
+    if (typeof version === 'function') {
+      registerRoutesCallback = version;
+      inner.version = config.version || require('./package.json').version;
+    } else if (version) {
+      inner.version = config.version || version;
```

Comment:
I think it's important that we document this prominently: I'd expect the explicit version I pass in to override whatever is the config, rather than the other way around.